parm/config/config.base.emc.dyn

```diff
@@ -154,46 +154,47 @@ export DO_COUPLED="NO"
 export DO_WAVE="NO"
 export DO_OCN="NO"
 export DO_ICE="NO"
-export DO_AERO=@DO_AERO@
+export DO_AERO="NO"
 export CCPP_SUITE="FV3_GFS_v17_p8"
 export WAVE_CDUMP="" # When to include wave suite: gdas, gfs, or both
 export cplwav2atm=".false."
 
-aero_nems_string=""
-if [[ $DO_AERO = "YES" ]]; then
-    aero_nems_string="_aero"
-else
-    aero_nems_string=""
-fi
-
 case "${APP}" in
   ATM)
     echo "APP=ATM; will use defaults"
-    export confignamevarfornems="atm${aero_nems_string}"
+    export confignamevarfornems="atm"
+    ;;
+  ATMA)
+    export DO_AERO="YES"
+    export confignamevarfornems="atm_aero"
     ;;
   ATMW)
     export DO_COUPLED="YES"
     export DO_WAVE="YES"
     export WAVE_CDUMP="both"
-    export confignamevarfornems="leapfrog_atm${aero_nems_string}_wav"
+    export confignamevarfornems="leapfrog_atm_wav"
     ;;
-  S2S)
+  S2S*)
     export DO_COUPLED="YES"
     export DO_OCN="YES"
     export DO_ICE="YES"
     export CCPP_SUITE="FV3_GFS_v17_coupled_p8"
-    export confignamevarfornems="cpld${aero_nems_string}"
-    ;;
-  S2SW)
-    export DO_COUPLED="YES"
-    export DO_WAVE="YES"
-    export DO_OCN="YES"
-    export DO_ICE="YES"
-    export CCPP_SUITE="FV3_GFS_v17_coupled_p8"
-    export WAVE_CDUMP="both"
-    export cplwav2atm=".true."
-    export confignamevarfornems="cpld${aero_nems_string}_wave"
+    export confignamevarfornems="cpld"
+
+    if [[ $APP =~ 'A$' ]]; then
+        export DO_AERO="YES"
+        export confignamevarfornems="${confignamevarfornems}_aero"
+    fi
+
+    if [[ $APP =~ '^S2SW' ]]; then
+        export DO_WAVE="YES"
+        export WAVE_CDUMP="both"    
+        export cplwav2atm=".true."
+        export confignamevarfornems="${confignamevarfornems}_wave"
+    fi
```

Comment:
Similarly, this is also not going to work:
```
DO_WAVE='unknown'
if [[ $APP =~ '^S2SW' ]]; then
  DO_WAVE="YES"
fi

echo "DO_WAVE is $DO_WAVE"
```
gives
```
DO_WAVE is unknown
```

Reply:
This one should work now too.